clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -831,13 +833,19 @@ public void testInboundConnectionsCountInConnectionCreationMetric() throws Excep
             for (int i = 0; i < conns; i++) {
                 Thread sender = createSender(serverAddress, randomPayload(1));
                 sender.start();
-                SocketChannel channel = ss.accept();
-                channel.configureBlocking(false);
-
-                selector.register(Integer.toString(i), channel);
+                threads.add(sender);
+                try (SocketChannel channel = ss.accept()) {
```

Comment:
if the exception is produced here, the following "join" won't be executed. Maybe we can use try-finally? for example:
```java
        try (ServerSocketChannel ss = ServerSocketChannel.open()) {
            ss.bind(new InetSocketAddress(0));
            InetSocketAddress serverAddress = (InetSocketAddress) ss.getLocalAddress();

            for (int i = 0; i < conns; i++) {
                Thread sender = createSender(serverAddress, randomPayload(1));
                sender.start();
                try (SocketChannel channel = ss.accept()) {
                    channel.configureBlocking(false);
                    selector.register(Integer.toString(i), channel);
                } finally {
                    sender.join();
                }
            }
        }
```

Reply:
Thanks @chia7712 
I think this suggestion is very insightful, I will follow it 😃